braindecode/datasets/datasets.py

```diff
@@ -0,0 +1,446 @@
+            all_windows_ds.append(windows_ds)
+        super().__init__(all_windows_ds)
+        self.info = info
+
+    # TODO: remove duplicate code here and in TUHAbnormal
+    # TODO: Create another class 'SpecificDataset'?
+    def split(self, some_property=None, split_ids=None):
+        """
+        Split the dataset based on some property listed in its info DataFrame
+        or based on indices.
+
+        Parameters
+        ----------
+        some_property: str
+            some property which is listed in info DataFrame
+        split_ids: list(int)
+            list of indices to be combined in a subset
+
+        Returns
+        -------
+        splits: dict{split_name: subset}
+            mapping of split name based on property or index based on split_ids
+            to subset of the data
+
+        """
+        assert split_ids is None or some_property is None, (
+            "can split either based on ids or based on some property")
+        if split_ids is None:
+            split_ids = _split_ids(self.info, some_property)
+        else:
+            split_ids = {split_i: split
+                         for split_i, split in enumerate(split_ids)}
+        return {split_name: Subset(self, split)
+                for split_name, split in split_ids.items()}
+
+
+def _split_ids(df, some_property):
+    assert some_property in df
+    split_ids = {}
+    for group_name, group in df.groupby(some_property):
+        split_ids.update({group_name: list(group.index)})
+    return split_ids
+
+
+class BNCI2014001(MOABBDataset):
+    """
+    see moabb.datasets.bnci.BNCI2014001
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__("BNCI2014001", *args, **kwargs)
+
+
+class HGD(MOABBDataset):
+    """
+    see moabb.datasets.schirrmeister2017.Schirrmeister2017
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__("Schirrmeister2017", *args, **kwargs)
+
+
+class TUHAbnormal(ConcatDataset):
+    """
+    Temple University Hospital (TUH) Abnormal EEG Corpus.
+
+    Parameters
+    ----------
+    trial_start_offset_samples: int
+        start offset from original trial onsets in samples
+    trial_stop_offset_samples: int
+        stop offset from original trial onsets in samples
+    supercrop_size_samples: int
+        supercrop size
+    supercrop_stride_samples: int
+        stride between supercrops
+    drop_samples: bool
+        whether or not have a last overlapping supercrop/window, when
+        supercrops/windows do not equally devide the continuous signal
+    target: str
+
+    mapping: dict{target_value: int}
+        maps target values to integers
+    """
+
+    def __init__(self, path, trial_start_offset_samples,
+                 trial_stop_offset_samples, supercrop_size_samples,
+                 supercrop_stride_samples, subject_ids=None,
+                 drop_samples=False, target="pathological", mapping=None):
+        windower = FixedLengthWindower(
```

Comment:
I haven't used the TUH dataset yet, but could someone want to make windows based on Events instead of sliding windows? In either case, I think it would be simpler to ask for a Windower instead of passing all the arguments directly (again, there could be a default).

Reply:
There are no events defined for the TUH data, so I wouldn't know how to do it. Again, I agree, there should be a default for the windower, leaving the possibility to the user to do whatever he wants.